### PR TITLE
Initial pass on context data

### PR DESCRIPTION
**Pull request by mitsuhiko**
Refs #3510
- Adds 'contexts' attribute to payloads[1]
- Adds Browser, Device, Runtime, and OS context schemas
- Autofill Browser/OS context from javascript's useragents
- Disable sentry_useragents for javascript/cocoa
- Show summary of contexts[2] for javascript/cocoa
- Disable gravatars for user contexts
- Push user into context chunks, and lower priority below event entries[3]

[1] context schemas

```
"contexts": {
    "device": {
        "name": "iPad",
        "model": "iPad5,1"
    },
    "os": {
        "name": "iOS",
        "version": "8.1"
    }
}
```

[2] context summary

screenshot 2016-06-27 10 36 00

[3] expanded contexts

screenshot 2016-06-27 10 49 50

@getsentry/infrastructure  @getsentry/cocoa @getsentry/ui

### CODE REVIEW

**Comment by dcramer** [comment]
IMO we should break up the context file. This has gotten to be a pattern lately and it's making some of the modules unwieldy. I dont know how deep we'll go into custom rendering for various styles of context, but as soon as these become more than "here are the field names" it really sucks to have to dig into a giant file to make changes.

tl;dr I'm a big proponent of "one class per file" in most situations

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -711,6 +712,7 @@ def create_partitioned_queues(name):
     'sentry.interfaces.Csp': 'sentry.interfaces.csp.Csp',
     'sentry.interfaces.AppleCrashReport': 'sentry.interfaces.applecrash.AppleCrashReport',
     'sentry.interfaces.Breadcrumbs': 'sentry.interfaces.breadcrumbs.Breadcrumbs',
+    'sentry.interfaces.Contexts': 'sentry.interfaces.contexts.Contexts',
```
Comment (dcramer) [inline src/sentry/conf/server.py]: I wouldn't add the full path

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -701,6 +701,7 @@ def create_partitioned_queues(name):
     'user': 'sentry.interfaces.user.User',
     'applecrashreport': 'sentry.interfaces.applecrash.AppleCrashReport',
     'breadcrumbs': 'sentry.interfaces.breadcrumbs.Breadcrumbs',
+    'contexts': 'sentry.interfaces.contexts.Contexts',
```
Comment (dcramer) [inline src/sentry/conf/server.py]: I'm not sure if we need to call it the plural form (context to me implies any amount of context), but I dont have strong feelings either way.

File: src/sentry/interfaces/contexts.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        for alias, inst in self._data.iteritems():
+            rv[alias] = inst.to_json()
+        return rv
+
+    def iter_tags(self):
+        for inst in self.iter_contexts():
+            for tag in inst.iter_tags():
+                yield tag
+
+    def get_path(self):
```
Comment (dcramer) [inline src/sentry/interfaces/contexts.py]: I'd use whatever the alias is going to be for the full path here

File: src/sentry/interfaces/contexts.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    indexed_fields = []
+
+    def __init__(self, alias, data):
+        self.alias = alias
+        self.data = data
+
+    def to_json(self):
+        return self.data
+
+    def flatten_index_value(self, value):
```
Comment (dcramer) [inline src/sentry/interfaces/contexts.py]: whats this mean? might be worthwhile to add some docstrings on the base type to explain input/output

**Comment by dcramer** [comment]
We're still going to tweak the UI, but leaving a few notes on that here:

- If we keep it a tabular grid (or if we have one, which we likely will) we probably should push it down below entries. This will including pushing user below, but arguably is how it should be.

- We should consider a summary view of the context data towards the top of the page, in a small compact space. The alternative is to throw it into the sidebar, but we haven't figured out a good solution to ensure thats representative yet.

- We might want to consider (can be a followup) having aggregate-wide information for these things, beyond tags. For example, if we start capturing more hardware stats, it'd be nice to have a breakdown of "90% of these errors had <50mb of free memory".

**Comment by dcramer** [comment]
Some notes:

- Always three summary context boxes
- Which three boxes depends on "experience" (web server, web client, desktop, mobile)
- When data isnt available we show ? + "Unknown X" + instructions if applicable (user)

Experiences:

- Web (Server): User, Runtime, OS
- Web (Client): User, Browser, OS
- Desktop: User, Device, OS (TBD)
- Mobile: User, Device, OS

In the future it's possible we'll make these configurable, or we'll just do it based on what's available.

**Comment by dcramer** [comment]
Here's where we're at today:

https://www.dropbox.com/s/jqaw2jkrmrvp4g7/Screenshot%202016-06-27%2010.36.00.png?dl=0
